src/main/kotlin/nl/avisi/kotlinwebtest/jdbc/Jdbc.kt

```diff
@@ -0,0 +1,149 @@
+import nl.avisi.kotlinwebtest.expressions.findExpressions
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.sql.DriverManager
+import java.sql.ResultSet
+import java.sql.SQLException
+import java.util.ArrayList
+import java.util.HashMap
+
+
+class JdbcStepRequest(var body: String? = null) : StepRequest, DatabaseRequest() {
+
+    override infix fun url(url: String) {
+        testStep.url = url
+    }
+
+    override infix fun driver(driver: String) {
+        testStep.driver = driver
+    }
+
+    lateinit var testStep: JdbcTestStep
+}
+
+class JdbcStepResponse(override val resultSet: List<HashMap<String, String>>?,
+                       override val success: Boolean,
+                       override val message: String? = null) : DatabaseStepResponse
+
+class JdbcTestStep(testCase: TestCase) : TestStep<JdbcStepRequest, JdbcStepResponse>(testCase, JdbcStepRequest()) {
+    var url: String? = null
+    var driver: String? = null
+
+    init {
+        request.testStep = this
+    }
+
+    fun resolveUrl(configuration: JdbcTestConfiguration): String? =
+            url ?: configuration.defaults.url
+
+    fun resolveDriver(configuration: JdbcTestConfiguration): String? =
+            driver ?: configuration.defaults.driver
+}
+
+class JdbcExecutor : Executor<JdbcTestStep> {
+
+    override fun execute(step: JdbcTestStep, executionContext: ExecutionContext): StepResponse {
+        val query = (step.request.query ?: error("No query configured for JDBC test step."))
+                .let { interpolateExpressions(it, executionContext) }
+
+        val configuration = executionContext.configuration[JdbcTestConfiguration::class]
+        val url = step.resolveUrl(configuration) ?: error("No url configured for JDBC test step.")
+        val driver = step.resolveDriver(configuration) ?: error("No driver configured for JDBC test step.")
+
+        var result: List<HashMap<String, String>>? = null
+
+        try {
+            Class.forName(driver)
+            log.info("Sending request:\n Querry: {}", query)
+            result = DriverManager.getConnection(url).use {
+                it.createStatement().use {
+                    resultSetToList(it.executeQuery(query))
+                }
+            }
+            log.info("Response: ", result)
+        } catch (e: ClassNotFoundException) {
+            log.error("Where is your Driver? Include in your pom!")
+            return JdbcStepResponse(result, false, "ClassNotFoundException: ${e.message}").also {
+                with(executionContext) {
+                    previousRequest = step.request
+                }
+            }
+        } catch (e: SQLException) {
+            log.error("SQL exception:", e)
+            return JdbcStepResponse(result, false, "SQLException request failed: ${e.message}").also {
+                with(executionContext) {
+                    previousRequest = step.request
+                }
+            }
+        } catch (e: IOException) {
```

Comment:
these catch blocks contain quite some duplication. Only the message differs. Kotlin does not support multi-catch (yet), unfortunatly. Maybe something like this:

```
val errorResponse = { exception -> 
log.error("JDBC request failed.", e)
JdbcStepResponse(result, false, "${e.javaClass.simpleName}: ${e.message}").also {
                with(executionContext) {
                    previousRequest = step.request
                }
            } }
}
```

```
try {
  // stuff here
} catch (e: IOException) {
  return errorResponse(e)
} catch (e: SQLException) {
  return errorResponse(e)
} catch (e: ClassNotFoundException) {
  return errorResponse(e)
}
```

(didn't write this in an Editor so probably doesn't compile)